actors/migration/nv8/test/modify_pending_proposals_test.go

```diff
@@ -0,0 +1,226 @@
+	dealStart := abi.ChainEpoch(252)
+	var deals []*market2.PublishStorageDealsReturn
+	for i := 0; i < 100; i++ {
+		var err error
+		v, err = v.WithEpoch(v.GetEpoch() + 1)
+		require.NoError(t, err)
+		deals = append(deals, publishDeal(t, v, worker, addrs[1+i%9], minerAddrs.IDAddress, fmt.Sprintf("deal%d", i),
+			1<<26, false, dealStart, 210*builtin.EpochsInDay))
+	}
+
+	// run migration
+	nextRoot, err := nv8.MigrateStateTree(ctx, v.Store(), v.StateRoot(), v.GetEpoch(), nv8.Config{})
+	require.NoError(t, err)
+
+	lookup := map[cid.Cid]runtime.VMActor{}
+	for _, ba := range exported.BuiltinActors() {
+		lookup[ba.Code()] = ba
+	}
+
+	v3, err := vm3.NewVMAtEpoch(ctx, lookup, v.Store(), nextRoot, v.GetEpoch()+1)
+	require.NoError(t, err)
+
+	// add 10 more deals after migration
+	for i := 0; i < 10; i++ {
+		var err error
+		v3, err = v3.WithEpoch(v.GetEpoch() + 1)
+		require.NoError(t, err)
+		deals = append(deals, publishV3Deal(t, v3, worker, addrs[1+i%9], minerAddrs.IDAddress, fmt.Sprintf("deal1%d", i),
+			1<<26, false, dealStart, 210*builtin.EpochsInDay))
+	}
+
+	// add even deals to a sector we will commit (to let the odd deals expire)
+	var dealIDs []abi.DealID
+	for i := 0; i < len(deals); i += 2 {
+		dealIDs = append(dealIDs, deals[i].IDs[0])
+	}
+
+	// precommit capacity upgrade sector with deals
+	sectorNumber := abi.SectorNumber(100)
+	sealedCid := tutil.MakeCID("100", &miner3.SealedCIDPrefix)
+	sealProof := abi.RegisteredSealProof_StackedDrg32GiBV1_1
+	preCommitParams := miner3.PreCommitSectorParams{
+		SealProof:     sealProof,
+		SectorNumber:  sectorNumber,
+		SealedCID:     sealedCid,
+		SealRandEpoch: v.GetEpoch() - 1,
+		DealIDs:       dealIDs,
+		Expiration:    v.GetEpoch() + 220*builtin.EpochsInDay,
+	}
+	vm3.ApplyOk(t, v3, addrs[0], minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.PreCommitSector, &preCommitParams)
+
+	// advance time to min seal duration
+	proveTime := v3.GetEpoch() + miner3.PreCommitChallengeDelay + 1
+	v3, _ = vm3.AdvanceByDeadlineTillEpoch(t, v3, minerAddrs.IDAddress, proveTime)
+
+	// Prove commit sector after max seal duration
+	v3, err = v3.WithEpoch(proveTime)
+	require.NoError(t, err)
+	proveCommitParams := miner3.ProveCommitSectorParams{
+		SectorNumber: sectorNumber,
+	}
+	vm3.ApplyOk(t, v3, worker, minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.ProveCommitSector, &proveCommitParams)
+
+	// In the same epoch, trigger cron to validate prove commit
+	// This activates deals hitting the pending deals structure.
+	vm3.ApplyOk(t, v3, builtin.SystemActorAddr, builtin.CronActorAddr, big.Zero(), builtin.MethodsCron.EpochTick, nil)
+
+	// Run cron after deal start to expire pending deals
+	v3, err = v3.WithEpoch(dealStart + 1)
+	require.NoError(t, err)
+	vm3.ApplyOk(t, v3, builtin.SystemActorAddr, builtin.CronActorAddr, big.Zero(), builtin.MethodsCron.EpochTick, nil)
+
+	// assert that power total invariants are broken
```

Comment:
This looks like out of date copy paste that can be fixed here and whereever you lifted it from